std/conv.d

```diff
@@ -5338,3 +5338,229 @@ unittest
+}
+
+unittest
+{
+    import std.ascii;
+    // empty/whites
+    static assert( !"".isHexLitteral);
+    static assert( !" \r".isHexLitteral);
+    static assert( !whitespace.isHexLitteral);
+    static assert( !""w.isHexLitteral);
+    static assert( !" \r"w.isHexLitteral);
+    static assert( !""d.isHexLitteral);
+    static assert( !" \r"d.isHexLitteral); 
+    // odd x strings
+    static assert( !("5" ~ whitespace).isHexLitteral);
+    static assert( !"123".isHexLitteral);
+    static assert( !"1A3".isHexLitteral);
+    static assert( !"1 23".isHexLitteral);
+    static assert( !"\r\n\tC".isHexLitteral);
+    static assert( !"123"w.isHexLitteral);
+    static assert( !"1A3"w.isHexLitteral);
+    static assert( !"1 23"w.isHexLitteral);
+    static assert( !"\r\n\tC"w.isHexLitteral);
+    static assert( !"123"d.isHexLitteral);
+    static assert( !"1A3"d.isHexLitteral);
+    static assert( !"1 23"d.isHexLitteral);
+    static assert( !"\r\n\tC"d.isHexLitteral);
+    // even x strings with invalid charset
+    static assert( !"12gG".isHexLitteral);
+    static assert( !"2A  3q".isHexLitteral);
+    static assert( !"12gG"w.isHexLitteral);
+    static assert( !"2A  3q"w.isHexLitteral);
+    static assert( !"12gG"d.isHexLitteral);
+    static assert( !"2A  3q"d.isHexLitteral);
+    // valid x strings
+    static assert( ("5A" ~ whitespace).isHexLitteral);
+    static assert( ("5A 01A C FF de 1b").isHexLitteral);
+    static assert( ("0123456789abcdefABCDEF").isHexLitteral);
+    static assert( (" 012 34 5 6789 abcd ef\rAB\nCDEF").isHexLitteral);
+    static assert( ("5A 01A C FF de 1b"w).isHexLitteral);
+    static assert( ("0123456789abcdefABCDEF"w).isHexLitteral);
+    static assert( (" 012 34 5 6789 abcd ef\rAB\nCDEF"w).isHexLitteral);
+    static assert( ("5A 01A C FF de 1b"d).isHexLitteral);
+    static assert( ("0123456789abcdefABCDEF"d).isHexLitteral);
+    static assert( (" 012 34 5 6789 abcd ef\rAB\nCDEF"d).isHexLitteral);
+    // library version allows what's pointed by issue 10454
+    static assert( ("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF").isHexLitteral);
+}
+
+/**
+The $(D hexString) function is intended to replace the hexadecimal literal strings
+starting with $(D 'x'), which could be removed to simplify the core language.
+
+The function takes a string made of hexadecimal digits and it returns
+the matching $(D string) by converting each pair of digit to a character.
+The input string can also include some ASCII white characters, which can be used
+to keep the literal string readable in the source code.
+
+Params:
+hexData = the $(D string) to be converted.
+C = character type of the output string. By default the function returns a string
+but can also return a $(D wstring) or a $(D dstring) if the parameter is set to
+$(D wchar) or $(D dchar).
+ */
+@property @safe nothrow pure
+auto hexString(string hexData, C = char)()
+if (hexData.isHexLitteral && isSomeChar!C)
+{
+    return hexStrImpl!(C)(hexData);
```

Comment:
Nitpick; omit parantheses for single-token template arguments: `return hexStrImpl!C(hexData);`